systemd/init

```diff
@@ -0,0 +1,52 @@
+#!/bin/busybox sh
+
+# Init for Wolfi OS
+# - Environment variables:
+#     VERBOSE=1 for debugging
+#     CMDLINE=/path/to/file to override /proc/cmdline, useful for testing
+# - Mount /proc, /sys, /dev if not already mounted
+# - Allow for override of path to /init with init=/path/to/init on kernel /proc/cmdline
+# - Allow for uuid or device path for root filesystem with root= on kernel /proc/cmdline
+# - Mount root, if specified and found, switch_root to it, and run systemd as init
+# - Otherwise, try to run systemd as init in the initrd
+
+# Verbose debugging
+info() { [ "$VERBOSE" = "1" ] && printf "INFO: $@\n" 1>&2; }
+
+# Override /proc/cmdline path as environment variable, useful for testing
+[ -z "$CMDLINE" ] && CMDLINE="/proc/cmdline"
+info "cmdline = [$(cat $CMDLINE)]"
+
+# Setup our system mounts if needed
+mkdir -p /proc /sys /dev
+findmnt /proc >/dev/null 2>&1 || mount -t proc none /proc
+findmnt /sys >/dev/null  2>&1 || mount -t sysfs none /sys
+findmnt /dev >/dev/null  2>&1 || mount -t devtmpfs none /dev
+
+# Allow override of path to init on kernel command line
+if (grep -qs "init=" $CMDLINE); then
+	init=$(sed -e "s:.*\binit=::" -e "s: .*::" $CMDLINE)
+	if [ ! -z "$init" ] && [ -x "$init" ]; then
+		# Exec init as overriden on the kernel command line
+		info "Running [$init] as init"
+		exec $init
```

Comment:
you're not honoring init= in if there is a 'root='

Reply:
Great catch.  You're absolutely right.

Reply:
Okay, fixed.